### DISCUSSION TITLE
HomeAssistant VS intellisense?

### DISCUSSION THREAD

**Discussion by jonathanrelf** [anhaehne/hhnl.HomeAssistantNet#55] [General]
Hi, 
So, all seems well and the demo automation compiles and runs as expected as I start to play with HomeAssistantNet.
However, in both Visual Studio Code & Visual Studio for Mac, the (presumably reflected?) HomeAssistant namespace doesn't offer auto-complete / intellisense? 

It appears incorrect in the editor but will compile correctly with dotnet build. 

Am I doing something wrong? I'd appreciate being able to get VS Code to know what's available in the HomeAssistant namespace because I'm unsure of all the things I could choose from, and it'd be preferable to having to switch to documentation or work by trial & error?

![image](https://user-images.githubusercontent.com/584662/158450550-d12dd413-5305-463a-9767-1a9ca2b18038.png)

**Comment by jonathanrelf** [comment]
@anhaehne Any thoughts?

**Reply by anhaehne** [reply]
I think this is just missing support in omni sharp. I've never tested this with vs code.

**Reply by jonathanrelf** [reply]
Thank you for saying OmniSharp, @anhaehne. I searched for 'OmniSharp Visual Studio Code' and found this link:
https://github.com/OmniSharp/omnisharp-vscode#using-net-6-builds-of-omnisharp

By navigating to Visual Studio Code Settings I could enable 'Omnisharp: Use Modern Net' which downloads the .NET 6 compatible version of OmniSharp. Whilst it restarted OmniSharp server, I found I had to restart VS Code to get the installation to work properly. 
![image](https://user-images.githubusercontent.com/584662/160198716-1ab3833d-ae41-4b3c-a36e-31ae55704a09.png)


And lo and behold! I can now get the Intellisense I need!
![image](https://user-images.githubusercontent.com/584662/160198040-8005afe3-6b45-4a61-90ab-482f52a0eb21.png)

I suspect as .NET 6 becomes the norm for all tools this will get easier. The Visual Studio for Mac still doesn't support .NET 6 natively which is being trialled in the Visual Studio for Mac Preview. Hence why I've been using VS Code for development on my Mac. 

Thanks for the continued support and I shall now be using the library in earnest!

Cheers, 
Jonathan
